extensions/guacamole-auth-ldap/src/main/java/org/apache/guacamole/auth/ldap/connection/ConnectionService.java

```diff
@@ -166,14 +194,52 @@
                     config.setProtocol(protocol.getString());
                 }
                 catch (LdapInvalidAttributeValueException e) {
-                    logger.error("Invalid value of the protocol entry: {}",
-                            e.getMessage());
+                    logger.error("Invalid value of the protocol entry: {}", e.getMessage());
                     logger.debug("LDAP exception when getting protocol value.", e);
                     return null;
                 }
+                
+                // Get proxy configuration, if any
+                GuacamoleProxyConfiguration proxyConfig;
+                try {
+                    
+                    // Get default proxy configuration values
+                    proxyConfig = LocalEnvironment.getInstance().getDefaultGuacamoleProxyConfiguration();
+                    String proxyHostname = proxyConfig.getHostname();
+                    int proxyPort = proxyConfig.getPort();
+                    EncryptionMethod proxyEncryption = proxyConfig.getEncryptionMethod();
+                    
+                    // Get the proxy hostname
+                    Attribute proxyHostAttr = entry.get(LDAP_ATTRIBUTE_PROXY_HOSTNAME);
+                    if (proxyHostAttr != null && proxyHostAttr.size() > 0)
+                        proxyHostname = proxyHostAttr.getString();
+                    
+                    // Get the proxy port
+                    Attribute proxyPortAttr = entry.get(LDAP_ATTRIBUTE_PROXY_PORT);
+                    if (proxyPortAttr != null && proxyPortAttr.size() > 0)
+                        proxyPort = Integer.parseInt(proxyPortAttr.getString());
+                    
+                    // Get the proxy encryption method
+                    Attribute proxyEncryptionAttr = entry.get(LDAP_ATTRIBUTE_PROXY_ENCRYPTION);
+                    if (proxyEncryptionAttr != null && proxyEncryptionAttr.size() > 0)
+                        proxyEncryption = EncryptionMethod.valueOf(proxyEncryptionAttr.getString());
```

Comment:
This will throw an `IllegalArgumentException` if the provided value does not identically match one of the internal names of the enum values (currently `NONE` and `SSL`).